public/app/core/components/Select/FolderPicker.test.tsx

```diff
@@ -42,6 +43,37 @@ describe('FolderPicker', () => {
     expect(pickerOptions[0]).toHaveTextContent('Dash 1');
     expect(pickerOptions[1]).toHaveTextContent('Dash 3');
   });
+
+  it('should allow creating a new option', async () => {
+    const newFolder = { title: 'New Folder', id: 3 } as DashboardSearchHit;
+
+    jest
+      .spyOn(api, 'searchFolders')
+      .mockResolvedValue([
+        { title: 'Dash 1', id: 1 } as DashboardSearchHit,
+        { title: 'Dash 2', id: 2 } as DashboardSearchHit,
+      ]);
+
+    const onChangeFn = jest.fn();
+
+    const create = jest.spyOn(api, 'createFolder').mockResolvedValue(newFolder);
+
+    render(<FolderPicker onChange={onChangeFn} enableCreateNew={true} allowEmpty={true} />);
+    expect(await screen.findByTestId(selectors.components.FolderPicker.containerV2)).toBeInTheDocument();
+
+    await userEvent.type(screen.getByLabelText('Select a folder'), newFolder.title);
+    await screen.findByText('Hit enter to add');
+
+    await userEvent.click(screen.getByText('Hit enter to add'));
+    await waitFor(() => {
+      expect(create).toHaveBeenCalledWith({ title: newFolder.title });
+    });
+
+    expect(onChangeFn).toHaveBeenCalledWith({ title: newFolder.title, id: newFolder.id });
+    await waitFor(() => {
+      expect(screen.getByText(newFolder.title)).toBeInTheDocument();
```

Comment:
Not too happy about this assertion but I couldn't find another one that worked reliably with `react-select`

Reply:
We have a [react-select-event](https://testing-library.com/docs/ecosystem-react-select-event/) library already installed. It helps to deal with `react-select` testing. Maybe it could help

Reply:
I had a quick look at the README and API and it looks like they don't have any custom assertions, in fact the README also use `expect(getByText("...")).toBeInTheDocument()` as an example.